File: prime not prime/python/all_prime_till.py
def if_prime(mm):
    if mm < 2 or (mm > 2 and mm % 2 == 0):
        return False
    till = int(mm ** 0.5) + 2
    for i in range(3, till, 2):
        if mm % i == 0 and mm != i:
            return False
    return True

"""all_prime_till is simple function which follow some simple algorithm to find
prime numbers till n(n is in count). in this function we mark a number and give it as argument in if_prime function which will tell us if the number is prime or not.

as we know, without 2 anyother even number is not prime. so we take only odd numbers as argument in all_prime function. """

def all_prime(n : int) -> list:
    prime_number = []  # result list ,contains only prime numbers from 2 to n
    if n < 2:
        return prime_number
    prime_number.append(2)
    for m in range(3, n + 1):
        if if_prime(m) == True:
            prime_number.append(m)
    return prime_number


print(*all_prime(1000000)) # example input

File: prime not prime/c++/all_prime_till.cpp
#include<vector>
#include<cmath>
#include<iostream>


bool prime(long long n)
{
    if(n < 2 || (n > 2 && n % 2 == 0)){
        return false;
    }
    long long till {std::sqrt(n) + 2};
    for(long long i = 3; i < till; i += 2){
        if( n % i == 0){
            return false;
        }
    }
    return true;
}
/*this function will return a list of prime number between 2 to n.*/
std::vector<long long> all_prime_till(long long n)
{
    std::vector<long long> res;
    if(n < 2){
        return res;
    }
    res.push_back(2);
    for(long long i = 3; i <= n; i += 2){
        if(prime(i) == true){
            res.push_back(i);
        }
    }
    return res;
}

int main()
{

    long long n{1000}; // n = 1000 as input
    std::vector<long long> prime_number_list = all_prime_till(n); // assigning result

    //output sector
    for(long long i = 0; i < prime_number_list.size(); i ++){
        std::cout << prime_number_list[i] << ", ";
    }
    return 0;
}

